### PR TITLE
feat: Flatpak lockdown

### PR DESCRIPTION
**Pull request by RKNF404**

### CODE REVIEW

**Comment by RoyalOughtness** [comment]
lgtm, lmk when this is ready for review

File: files/justfiles/hardening.just [hunk 1]
```diff
@@ -14,6 +14,88 @@ harden-flatpak FLATPAK="":
         flatpak override --user --env=LD_PRELOAD=/var/run/host/usr/lib64/glibc-hwcaps/x86-64-v"$bestuarch"/libhardened_malloc.so $flatpak_id
     fi
 
+# Harden Flatpaks further by disabling all permissions by default and rejecting known arbitrary DBus names
+flatpak-permissions-lockdown:
+    #!/usr/bin/bash
+    kCommand="flatpak override --user"
```
Comment (RoyalOughtness) [inline files/justfiles/hardening.just:20]: since this is only for user flatpaks, can you state this in the comment above

Comment (RKNF404) [reply files/justfiles/hardening.just:20]: is what I added good?

Comment (RoyalOughtness) [reply files/justfiles/hardening.just:20]: yup

**Review by RoyalOughtness — DISMISSED** [review]

File: files/justfiles/hardening.just [hunk 2]
```diff
@@ -14,6 +14,88 @@ harden-flatpak FLATPAK="":
...
+    echo "-- System Bus Name Access --"
+    for i in ${kKnownSystemBusNames[@]}; do
+        echo "	Rejecting $i..."
+        $kCommand --system-no-talk-name=$i
+    done
+    echo ""
+    echo "-- Peristent Filesystem Grant --"
+    echo "Note: This is to unbreak many Flatpaks by allowing the app to store persistent data in their own, isolated home directory without accessing the user's"
+    echo "	Granting access to peristent home..."
+    $kCommand --perist=.
```
Comment (TheHooly) [inline files/justfiles/hardening.just]: Typo: Should be `--persist=.`

File: files/justfiles/hardening.just [hunk 3]
```diff
@@ -14,6 +14,88 @@ harden-flatpak FLATPAK="":
...
+    echo ""
+    echo "-- Granting Flatseal Access to Bus Names --"
+    for i in ${kFlatsealNameAccess[@]}; do
+        echo "	Rejecting $i..."
+        $kCommand --talk-name=$i com.github.tchx84.Flatseal
+    done
+    echo ""
+    echo "Done"
+
```
Comment (TheHooly) [inline files/justfiles/hardening.just:104]: `--device=dri` can and should be added by default, some apps rely on getting it via `device=all` and break without it. Practically all apps have and need this permission in some form.

**Review by TheHooly — CHANGES_REQUESTED** [review]

File: files/justfiles/hardening.just (hunk 2 above)
Comment (RKNF404) [reply files/justfiles/hardening.just]: Thanks, fixed

File: files/justfiles/hardening.just (hunk 3 above)
Comment (RKNF404) [reply files/justfiles/hardening.just:104]: lmk if there needs to be a better note for that section, otherwise it's done

**Review by TheHooly — APPROVED** [review]

**Review by RoyalOughtness — APPROVED** [review]

**Pull request merged**